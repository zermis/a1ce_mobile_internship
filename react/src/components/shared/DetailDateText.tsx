import React from 'react';
import { Text, StyleSheet, View } from 'react-native';
import Colors from '../../constants/Colors';

interface Props {
    header?: string;
    body?: string;
}

const DetailDateText = (props: Props) => {
    return (
        <View style={styles.flexBetweenInnerContainer}>
            <Text style={styles.textSubHeader}>{props.header}</Text>
            <Text style={styles.textBody}>{props.body}</Text>
        </View>
    );
};

const styles = StyleSheet.create({
    flexBetweenInnerContainer: {
        flexDirection: 'row',
        justifyContent: 'space-between',
        position: 'relative'
    },
    textSubHeader: {
        fontSize: 14,
        fontFamily: 'ibm-semibold',
        color: Colors.date
    },
    textBody: {
        textAlign: 'justify',
        fontSize: 14,
        fontFamily: 'ibm-regular',
        color: Colors.textBody
    }
});

export default DetailDateText;
