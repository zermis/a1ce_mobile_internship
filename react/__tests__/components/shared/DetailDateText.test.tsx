import DetailDateText from '../../../src/components/shared/DetailDateText';
import renderer from 'react-test-renderer';

it('render detail date text correctly', () => {
    const tree = renderer
        .create(<DetailDateText header="header" body="body" />)
        .toJSON();
    expect(tree).toMatchSnapshot();
});
