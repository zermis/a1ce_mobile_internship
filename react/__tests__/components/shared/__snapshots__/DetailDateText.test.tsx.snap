// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render detail date text correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-between",
      "position": "relative",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#A53638",
        "fontFamily": "ibm-semibold",
        "fontSize": 14,
      }
    }
  >
    header
  </Text>
  <Text
    style={
      Object {
        "color": "#848484",
        "fontFamily": "ibm-regular",
        "fontSize": 14,
        "textAlign": "justify",
      }
    }
  >
    body
  </Text>
</View>
`;
